src/components/Form/formFunctions.ts

```diff
@@ -87,89 +85,104 @@ const getErrorType = (
     return Errors.none;
 };
 
-export const onChangeText = (
-    fieldSetter: FieldSetter,
-    errorSetter: ErrorSetter,
+export const onChangeText = <
+    SpecificFields extends Fields,
+    SpecificErrors extends FormErrors<SpecificFields>,
```

Comment:
Does this second generic parameter real parameter? Can't we just have `errorSetter: Setter<FormErrors<SpecificFields>>` and so have one generic parameter?

(similarly in the other places)

Reply:
Okay - I think we still need SpecificErrors as a generic parameter for checkErrorOnSubmit because it has errors as an actual parameter, but let me know if there's something I could do to get rid of the generic parameter in this case?

Reply:
Does this work?

```ts
export const checkErrorOnSubmit = <SpecificFields extends Fields>(
    errorType: FormErrors<SpecificFields>,
    errorSetter: (errors: FormErrors<SpecificFields>) => void,
    keysToCheck?: string[]
): boolean => {
```

Reply:
No, because errorType isn't necessarily FormErrors<SpecificFields> it should be extends FormErrors<SpecificFields> I think
I get this error
```
Argument of type 'Dispatch<SetStateAction<Required<FormErrors<CollectionCentreFields>>>>' is not assignable to parameter of type '(errors: FormErrors<CollectionCentreFields>) => void'.
  Types of parameters 'value' and 'errors' are incompatible.
    Type 'FormErrors<CollectionCentreFields>' is not assignable to type 'SetStateAction<Required<FormErrors<CollectionCentreFields>>>'.
      Type 'FormErrors<CollectionCentreFields>' is not assignable to type 'Required<FormErrors<CollectionCentreFields>>'.
        Types of property 'name' are incompatible.
          Type 'Errors | undefined' is not assignable to type 'Errors'.
            Type 'undefined' is not assignable to type 'Errors'.
```